docs/guide/writing-migrations.md

```diff
@@ -0,0 +1,262 @@
+# Writing Migrations
+
+Migrations in ApostropheCMS allow you to make targeted changes to your database, ensuring that your data stays in sync with the evolving structure of your code. They are particularly useful when new code deployments introduce changes to the underlying data model or schema, such as adding new fields, removing deprecated properties, or correcting inconsistencies in existing content. In this guide, we will walk through how to write migrations to add or remove properties from existing pieces and widgets.
```

Comment:
In this coming release we are shipping logic to automatically add a default value for any new schema field that has a `def` property or is of a type that has its own `def`, which covers most types (for instance, for `string` the fallback `def` is `''`). So the need to write migrations just for a brand new field alone is about to drop off considerably.

Reply:
Doubling down on this because my work to automatically add default values has merged to `main` and release day is Wednesday, so it makes sense to address it here:

If your goal is to ensure a newly-added field in the schema will be present in the database for existing documents with its default value (as specified by `def`, or the fallback `def` of the field type such as the empty string for `string` fields), then you do not need to write a migration. As of version 4.x, this is automatic. However if you need to transform the existing content of the database in another way, such as renaming or removing a property or transforming a number to a string, et cetera, then migrations are the right tool for you.